protocol/messenger.go

```diff
@@ -4102,16 +4102,11 @@ func (m *Messenger) markAllRead(chatID string, clock uint64, shouldBeSynced bool
 	return m.persistence.SaveChats([]*Chat{chat})
 }
 
-func (m *Messenger) MarkAllRead(chatID string) error {
-	notifications, err := m.persistence.DismissAllActivityCenterNotificationsFromChatID(chatID, m.getCurrentTimeInMillis())
+func (m *Messenger) MarkAllRead(ctx context.Context, chatID string) error {
+	_, err := m.DismissAllActivityCenterNotificationsFromChatID(ctx, chatID, m.GetCurrentTimeInMillis())
 	if err != nil {
 		return err
 	}
-	err = m.syncActivityCenterNotifications(notifications)
```

Comment:
@qfrank in case of an activity like this one, we need to sync it, so marking all or particular notifications as read/unread manually is an action that should be synced.

Reply:
In this particular case all is good cause this is called when the user is marking all notifications coming from the chat with chatID as read and that's ok, but just wanted to say about the general approach, but it might be that I was not clear enough, please read this, more detailed [comment](https://github.com/status-im/status-go/pull/3979#issuecomment-1702403086).

Reply:
@qfrank It seems to me too that this is wrong here now. 
`Mark all as read` changes the state of the notification and this fact should be synced. And I don't see any syncing in `DismissAllActivityCenterNotificationsFromChatID`.

Or is it some new UI approach you was talking about?

Reply:
It's here: 
https://github.com/status-im/status-go/blob/95048404de53733dd1772b64d628c71a65434532/protocol/messenger_activity_center.go#L452

@igor-sirotin